src/arkit/ARKitDevice.js

```diff
@@ -60,6 +63,58 @@ export default class ARKitDevice extends XRDevice {
 			this._arKitWrapper = null;
 			this._arWatcher = null;
 		}
+
+		// hit test
+
+		this._hitTestSources = [];
+		this._hitTestResults = new Map();
+		this._hitTestResultsForNextFrame = new Map();
+
+		// Input set up
+		// Note: This touch input event is not AR kit specifi so that
+		// should we move this logic out of ARKitDevice?
+
+		this._gamepads = [];
+		this._gamepadInputSources = [];
+		this._touches = [];
+
+		const primaryButtonIndex = 0;
+		this._gamepads.push(createGamepad('', 'right', 1, true));
+		this._gamepadInputSources.push(new GamepadXRInputSource(this, {}, 0));
```

Comment:
What is GamepadXRInputSource? Is that private to the iOSViewer?

Reply:
It is in webxr-polyfill.js class. 

```
import GamepadXRInputSource from 'webxr-polyfill/src/devices/GamepadXRInputSource';
```

It isn't exposed to users. The polyfill manages inputSource and gamepad with it.